### DISCUSSION TITLE
Self assessment questions - suggested tweaks

### DISCUSSION THREAD

**Discussion by skyf0xx** [kinfoundation/kin-rfcs#160] [KRE 4.0 Discussion]
I think the following self assessment questions could be tweaked:

> Does your app integrate a fiat-to-crypto payment solution from Ramp, Simplex, Wyre or MoonPay which enables your end-users to purchase Kin?

I think there's no need to specify the payment service, as long as it is a Kin payment service.

> Do you permit end-users to withdraw 100% of their Kin balance from the app? (If you only enable partial withdrawals or you impose a withdrawal fee you must answer no).

As a dev It's usually a good idea to limit the withdrawal amount to a **certain extent**. For example,  an app may have a sign up bonus of 1,000 Kin for new users. This lets users play with Kin before needing to do earns. However, it also opens the app to 'farmers' who repeatedly create and delete accounts on the app for the sign on bonus and drain it.

To prevent these drains, a dev can prevent the user from withdrawing e.g. 3x of the sign on bonus. This means the user's account can never go below 3,000 Kin through withdrawals. However, they can still spend it by using it within the app 

This way, the bonus still motivates the user, and the user is unable to do constant farming attacks on the app

**Comment by hann228** [comment]
> I think there's no need to specify the payment service, as long as it is a Kin payment service.

We think to begin with it's better to be specific.   The questions are intended to be updated over time. The risk with making broad questions is they become open to interpretation and subsequently can be interpreted in a manner which is adverse to the objective of the question.
 
> As a dev It's usually a good idea to limit the withdrawal amount to a **certain extent**. For example, an app may have a sign up bonus of 1,000 Kin for new users. This lets users play with Kin before needing to do earns. However, it also opens the app to 'farmers' who repeatedly create and delete accounts on the app for the sign on bonus and drain it.

This is an interesting insight, we will look at this further and consider if any further tweaks are required.
